tls/s2n_record_read.c

```diff
@@ -46,7 +46,16 @@ int s2n_sslv2_record_header_parse(
 
     POSIX_GUARD(s2n_stuffer_read_uint8(in, record_type));
 
-    uint8_t protocol_version[S2N_TLS_PROTOCOL_VERSION_LEN];
+    /* The SSLv2 header is only 3 bytes (technically sometimes 2, but we only support 3 --
+     * see https://www.ietf.org/archive/id/draft-hickman-netscape-ssl-00.txt). So by reading
```

Comment:
Where do we fail if the format is wrong? Or do we not make it here?

Reply:
I'm reading the spec again and I think this comment is wrong. Let me rewrite.

Reply:
It was very wrong. Thanks for asking the question!